packages/SwingSet/src/kernel/state/vatKeeper.js

```diff
@@ -543,16 +543,17 @@ export function makeVatKeeper(
    * Store a snapshot, if given a snapStore.
    *
    * @param {VatManager} manager
+   * @param {boolean} [keepWorker]
    * @returns {Promise<void>}
    */
-  async function saveSnapshot(manager) {
+  async function saveSnapshot(manager, keepWorker) {
```

Comment:
I'd suggest flipping the polarity and calling the option `restartWorker`. For some reason "keepWorker" doesn't tell me what the two possibilities are as much as something with "restart" in the name. Like, if `keepWorker=false`, does it mean we're killing the worker entirely, and just relying upon the next delivery to page it back in? "restartWorker=true" tells me that we're going to kill it and start a new one right away, which means that not doing that might be an option. I dunno, maybe just me.

Also, not something we can do anything about now, but it's a bit weird that *vatKeeper*, being a frontend for the DB and in a subdirectory named `state/`, has an option that talks about how to manage child processes. Ideally vatKeeper would be only about state, and the relationship between vat-warehouse (the caller) and vatKeeper and the manager would let vatKeeper not take this option (and actually not take `manager` either). Like, maybe vat-warehouse gets a snapshotReadStream from the manager, and passes it into vatKeeper.saveSnapshot. Something to think about for later.

Reply:
I am very, very allergic to positional boolean parameters that default to true.

However I do see the possible confusion with "page out the worker".

Reply:
yeah, good point, I'll think about potential future cleanups for this, maybe I can find a word that triggers the right association in my mind but also has a good "false" default. (I'm still not used to JS being so casual about missing arguments, and haven't internalized the idea that omitting the argument is an accepted way of saying "false", which then obviously conflicts with needing a default of "true").